openwisp_radius/base/models.py

```diff
@@ -234,9 +234,12 @@ def clean(self):
 
 class UserAttributeValidationMixin(AttributeValidationMixin):
     def _get_validation_queryset_kwargs(self):
-        return dict(
-            organization=self.organization, user=self.user, attribute=self.attribute
-        )
+        # only add `organization` key if it exists
+        if getattr(self, 'organization_id'):
+            return dict(
+                organization=self.organization, user=self.user, attribute=self.attribute
+            )
+        return dict(user=self.user, attribute=self.attribute)
```

Comment:
there's a bit of duplication (redefinition of the same dict keys/values) and inefficiency (the call to `getattr(self, 'organization_id')` is wasted) here, a better way would be:

```python
kwargs = dict(user=self.user, attribute=self.attribute)
org = getattr(self, 'organization', None)
if org:
    kwargs['organization'] = org
return kwargs
```

Reply:
noted.